SpatialGDK/Source/SpatialGDK/Public/Utils/EntityFactory.h

```diff
@@ -1,39 +1,47 @@
 // Copyright (c) Improbable Worlds Ltd, All Rights Reserved
- 
+
 #pragma once
- 
+
 #include "SpatialCommonTypes.h"
 #include "Utils/SpatialStatics.h"
 
-#include <WorkerSDK/improbable/c_schema.h>
-#include <WorkerSDK/improbable/c_worker.h>
-
 DECLARE_LOG_CATEGORY_EXTERN(LogEntityFactory, Log, All);
 
 class AActor;
+class UAbstractLBStrategy;
 class USpatialActorChannel;
 class USpatialNetDriver;
 class USpatialPackageMap;
 class USpatialClassInfoManager;
 class USpatialPackageMapClient;
- 
+
 namespace SpatialGDK
 {
-class SpatialRPCService;	
+class InterestFactory;
+class SpatialRPCService;
 
 struct RPCsOnEntityCreation;
 using FRPCsOnEntityCreationMap = TMap<TWeakObjectPtr<const UObject>, RPCsOnEntityCreation>;
- 
+
 class SPATIALGDK_API EntityFactory
 {
 public:
 	EntityFactory(USpatialNetDriver* InNetDriver, USpatialPackageMapClient* InPackageMap, USpatialClassInfoManager* InClassInfoManager, SpatialRPCService* InRPCService);
- 
+
 	TArray<FWorkerComponentData> CreateEntityComponents(USpatialActorChannel* Channel, FRPCsOnEntityCreationMap& OutgoingOnCreateEntityRPCs, uint32& OutBytesWritten);
 	TArray<FWorkerComponentData> CreateTombstoneEntityComponents(AActor* Actor);
 
 	static TArray<Worker_ComponentId> GetComponentPresenceList(const TArray<FWorkerComponentData>& ComponentDatas);
 
+	static TArray<FWorkerComponentData> CreatePartitionEntityComponents(const Worker_EntityId EntityId, const InterestFactory* InterestFactory, const UAbstractLBStrategy* LbStrategy, VirtualWorkerId VirtualWorker);
+
+	static inline bool IsClientAuthoritativeComponent(Worker_ComponentId ComponentId)
+	{
+		return ComponentId == SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID
```

Comment:
Should we check here the contents of `REQUIRED_COMPONENTS_FOR_AUTH_SERVER_INTEREST` ?